autorest/adal/cli_token.go

```diff
@@ -0,0 +1,32 @@
+package adal
+
+import "strconv"
+
+type AzureCLIToken struct {
+	AccessToken      string `json:"accessToken"`
+	Authority        string `json:"_authority"`
+	ClientID         string `json:"_clientId"`
+	ExpiresIn        int     `json:"expiresIn"`
+	ExpiresOn        string `json:"expiresOn"`
+	IdentityProvider string `json:"identityProvider"`
+	IsMRRT           bool    `json:"isMRRT"`
+	RefreshToken     string `json:"refreshToken"`
+	Resource         string `json:"resource"`
+	TokenType        string `json:"tokenType"`
+	UserId           string `json:"userId"`
+}
+
+func AzureCLIAccessTokensPath() string {
+	return "~/.azure/accessTokens.json"
```

Comment:
Looks like the simplest way to do this is with [mitchellh/go-homedir](https://github.com/mitchellh/go-homedir) - I've pushed an update for this, and can confirm this works on macOS and Windows using this sample app:

```
$ cat main.go
package main

import "log"
import "github.com/mitchellh/go-homedir"

func main() {
  path, err := homedir.Expand("~/.azure/accessTokens.json")
  log.Printf("Path: %s", path)
  log.Printf("Error: %+v", err)
}
```

<img width="470" alt="screen shot 2017-08-22 at 13 05 58" src="https://user-images.githubusercontent.com/666005/29564653-a8a4c3d0-873a-11e7-8874-33362f10d953.png">

<img width="596" alt="screen shot 2017-08-22 at 13 04 06" src="https://user-images.githubusercontent.com/666005/29564642-a050326e-873a-11e7-8cdb-b1fe7e042776.png">